### PR TITLE
Sync README.rst and docs/summary.rst.

### PR DESCRIPTION
**Pull request by EnricoMi**
* Merges changes from #1119 of summary.rst into README.rst
* Merges changes from #1088 of README.rst into summary.rst
* Also considers #1158 and #1289

### CODE REVIEW

**Comment by EnricoMi** [comment]
You can check that `README.rst` and `docs/summary.rst` are in sync as follows:

```
diff <(sed -e "s/docs\///g" README.rst) docs/summary.rst
diff README.rst <(sed -E -e "s/<(.*\.rst)/<docs\/\1/g" docs/summary.rst)
```

The only difference surfaced is:
```
137a135
> .. inclusion-marker-tensorflow-start-do-not-remove
236a235
> .. inclusion-marker-tensorflow-end-do-not-remove
```

Because a second set of `inclusion-marker-tensorflow-start-do-not-remove` and `inclusion-marker-tensorflow-end-do-not-remove` markers exist in `summary.rst` but not `README.rst`, introduced in #1158. The two sets of those markers interleave. Not sure if this is expected. I can remove them with this PR if you like.

@sblotner, @IgorWilbert and @alsrgv might be interested in reviewing this.

File: docs/summary.rst [hunk 1]
```diff
@@ -48,11 +52,13 @@ about who's involved and how Horovod plays a role, read the LF AI `announcement
 
 .. contents::
 
+
+See the full documentation and an API reference at https://horovod.readthedocs.io/en/latest.
```
[the remark this answers is not in the document]
Comment (EnricoMi) [reply docs/summary.rst]: I have rephrased it to `The full documentation and an API reference are published at https://horovod.readthedocs.io/en/latest.` I would not like the sentence to start with an URL.

File: docs/summary.rst [hunk 2]
```diff
@@ -116,15 +121,14 @@ To compile Horovod from source, follow the instructions in the `Contributor Guid
...
 Horovod core principles are based on `MPI <http://mpi-forum.org/>`_ concepts such as *size*, *rank*,
-*local rank*, *allreduce*, *allgather* and, *broadcast*. See `this page <concepts.rst>`_ for more details.
+*local rank*, **allreduce**, **allgather** and, *broadcast*. See `this page <concepts.rst>`_ for more details.
 
 Supported frameworks
 --------------------
 See these pages for Horovod examples and best practices:
 
-- `Horovod with TensorFlow <tensorflow.rst>`_
+- `Horovod with TensorFlow <#usage>`__ (Usage section below)
```
[the remark this answers is not in the document]
Comment (EnricoMi) [reply docs/summary.rst]: done

**Comment by EnricoMi** [comment]
@sblotner regarding the duplicate `inclusion-marker-tensorflow-start-do-not-remove` marker in `docs/summary.rst`: is that expected or should I revert the pair introduced in #1158?. (see https://github.com/horovod/horovod/pull/1538#issuecomment-558568943 above)

File: docs/summary.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (sblotner) [reply docs/summary.rst]: Excellent revision, thanks!

**Comment by sblotner** [comment]
@EnricoMi , Regarding the duplicate inclusion-marker-tensorflow-start-do-not-remove marker in docs/summary.rst:, I need to dig deeper on that, I didn't implement it initially, so am not sure off the top of my head if it's required or not. I'll get back to you after I dig around a bit. Thank you!

**Comment by EnricoMi** [comment]
Thanks for looking into the `inclusion-marker`, it got introduced [here](https://github.com/horovod/horovod/pull/1158/files#diff-9a346253891bb898bdbe049a2f1a8d42R116) and [here](https://github.com/horovod/horovod/pull/1158/files#diff-9a346253891bb898bdbe049a2f1a8d42R187).

They are used in [docs/tensorflow.rst](https://github.com/horovod/horovod/pull/1158/files#diff-f995a0894b906686ac796fcaeaa12220). Looks like they are used to extract a bit from `summary.rst` into `tensorflow.rst`.

**Comment by tgaddair** [comment]
Hey @EnricoMi, can you rebase on master?  Looks like builds are failing due to not having [this](https://github.com/horovod/horovod/commit/c8c53a9c901e980e3f0a10634322580743bd17b8) fix.

**Comment by sblotner** [comment]
@EnricoMi Oops, I'm definitely the author of that change! Thanks for helping me remember :) 

This is used to help single source the TensorFow example/instructions to reduce copy/pasting. 

My recommendation is to keep the `inclusion-marker-tensorflow-start-do-not-remove` and `inclusion-marker-tensorflow-end-do-not-remove` in both files (README.rst and summary.rst)

Thank you for your attention to detail

**Comment by EnricoMi** [comment]
@sblotner Thanks for looking into the inclusion marker. I have noticed that I have described the situation incorrectly. Here is the correct current situation:
* There is a `inclusion-marker-start-do-not-remove` and `inclusion-marker-end-do-not-remove` in **both** files. I misread this as `inclusion-marker-tensorflow-...`.
* There is a `inclusion-marker-tensorflow-start-do-not-remove` and `inclusion-marker-tensorflow-end-do-not-remove` in `docs/summary.rst` only. It selects the tensor flow example to re-use it in `docs/tensorflow.rst`, as you described above.

I will copy them to `README.rst` as well, just to get those two files in sync. Then, diffing them should be empty:
```
diff <(sed -e "s/docs\///g" README.rst) docs/summary.rst
```

**Comment by EnricoMi** [comment]
@tgaddair @sblotner I think this is ready to merge into master.

**Comment by tgaddair** [comment]
Perfect, thanks @EnricoMi!

**Pull request merged**